core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -147,6 +124,28 @@ sealed abstract class Pull[+F[_], +O, +R] {
 
 object Pull extends PullLowPriority {
 
+  implicit class StreamPullOps[F[_], O](val pull: Pull[F, O, Unit]) extends AnyVal {
```

Comment:
Make `val` private or else this extension will allow syntax like `p.pull.pull.pull.pull.pull.pull`

Reply:
`value class parameter must be a val and not be private[this]`

Reply:
I can rename to `self`, if that helps...

Reply:
`private val self: Pull[F, O, Unit]` should be fine... Note the other implicit classes we have with a public `self` val have the same issue -- `foo.self.self.self.self.self`

Reply:
Ok... the problem is that this would mean allocating (you can no longer make them `AnyVal`).

Reply:
That's not right...

```scala
io/src/main/scala-2.13/fs2/io/CollectionCompat.scala
27:  implicit class JIteratorOps[A](private val self: java.util.Iterator[A]) extends AnyVal {
30:  implicit class JIterableOps[A](private val self: java.lang.Iterable[A]) extends AnyVal {
33:  implicit class JSetOps[A](private val self: java.util.Set[A]) extends AnyVal {
36:  implicit class ListOps[A](private val self: List[A]) extends AnyVal {
39:  implicit class SetOps[A](private val self: Set[A]) extends AnyVal {
42:  implicit class EnumerationOps[A](private val self: java.util.Enumeration[A]) extends AnyVal {
```

Reply:
🤔

Reply:
Ok, just pushed version with private and anyval. I got mistaken with the `private[this]` not allowed.